src/EA8AE0.c

```diff
@@ -1,29 +1,254 @@
 #include "dead.h"
 #include "common.h"
+#include "effects.h"
 
 #define NAMESPACE EA8AE0
 
 extern s32 D_80248388[];
+extern s32 D_80248380;
+
+typedef struct {
+    char unk_00[0x4];
+    f32 unk_04;
+    f32 unk_08;
+    f32 unk_0C;
+    f32 unk_10;
+    char unk_14[0x4];
+    s32 unk_18;
+    s32 unk_1C;
+    s32 unk_20;
+    X32 unk_24;
+    s32 unk_28;
+    s32 unk_2C;
+    s32 unk_30;
+    u8 unk_34;
+    s32 unk_38;
+    char unk_3C[0xC];
+    EffectUnkStruct1* unk_48;
+    char unk_4C[0x24];
+    s32 unk_70;
+    s32 unk_74;
+} EffectStruct;
+
+typedef struct temp {
+    char unk_00[0xC];
+    EffectStruct* unk_0C;
+} temp;
```

Comment:
Name this, and definitely don't typedef something called "temp"

Reply:
Again, copied over from Maide's decomp, I just kept it how it is for consistency sake and I also wouldn't really know how to properly name this since for now it only has one known field.

Reply:
I don't think naming these things is crucial at this stage, nor do I really see why typedefing it is bad